Tests/SwiftSyntaxBuilderTest/FunctionTests.swift

```diff
@@ -66,24 +40,25 @@ final class FunctionTests: XCTestCase {
   }
 
   func testParensEmittedForNoArgumentsAndNoTrailingClosure() {
-    let buildable = FunctionCallExpr(calledExpression: "test")
+    let buildable = FunctionCallExpr(calledExpression: Expr("test"))
     AssertBuildResult(buildable, "test()")
   }
 
   func testParensEmittedForArgumentAndTrailingClosure() {
-    let buildable = FunctionCallExpr(calledExpression: "test", trailingClosure: ClosureExpr()) {
+    let buildable = FunctionCallExpr(calledExpression: Expr("test"), trailingClosure: ClosureExpr()) {
```

Comment:
Hmm.. Not sure I like that we do this. 
I think that `FunctionCallExpr(calledExpression: "test", trailingClosure: ClosureExpr())` is a bit easier to to read and it's easier to get started with SwiftSyntaxBuilder

Reply:
I thought the same at first as well but after writing this a few times, I think I mildly prefer the new syntax. Here’s why:

- When you create a `FunctionCallExpr`, code completion will add you a placeholder that has type `ExprSyntaxProtocol`. The natural step to do next is to try and create something that’s `ExprSyntaxProtocol`, so you probably start with `Expr`. Knowing that you can also pass a string literal requires more in-depth knowledge of SwiftSyntaxBuilder, which newcomers probably won’t have
- In the new design, you can jump-to-definition on `Expr` to see how the parser is invoked. This makes it easier to understand what’s actually going on IMO because there are less magic conversions going on behind the scenes.
- This also allows more complex expressions (like member access). The old design was only correct for identifiers (because it created an `IdentifierExpr`)

Thinking about it some more, I think we could get this working again by changing the parameter type from `ExprSyntaxProtocol` to `ExprBuildable` where `ExprBuildable` is defined as follows:

```swift
protocol ExprBuildable {
  func createExpr() -> Expr
}

extension ExprSyntaxProtocol: ExprBuildable {
  func createExpr() -> Expr { return Expr(self) }
}

extension String: ExprBuildable {
  func createExpr() -> Expr { return Expr(self) }
}
```

That’s probably not too terrible. WDYT?

Reply:
OK, I just tried what I suggested above and it doesn’t work:

```swift
extension ExprSyntaxProtocol: ExprBuildable // error: Extension of protocol 'ExprSyntaxProtocol' cannot have an inheritance clause
```

Given that this simple approach doesn’t work, it would mean that we’d need to explicitly conform all `Expr` nodes go `ExprBuildable`, which defeats the beauty of the new implementation. Considering the arguments I gave above, I think that any implementation we find that makes it possible to pass strings here makes `SwiftSyntaxBuilder` harder to understand instead of easier.

Reply:
OK, I just tried what I suggested above and it doesn’t work:

```swift
extension ExprSyntaxProtocol: ExprBuildable // error: Extension of protocol 'ExprSyntaxProtocol' cannot have an inheritance clause
```

Given that this simple approach doesn’t work, it would mean that we’d need to explicitly conform all `Expr` nodes go `ExprBuildable`, which defeats the beauty of the new implementation. Considering the arguments I gave above, I think that any implementation we find that makes it possible to pass strings here makes `SwiftSyntaxBuilder` harder to understand instead of easier.

Reply:
I see. 
Let's go with the new design.
I think I also need to dive deeper into the usage of the new API. 😁